sdk/python/feast/feature_server.py

```diff
@@ -250,9 +255,8 @@ async def health():
     @app.post("/materialize", dependencies=[Depends(inject_user_details)])
     def materialize(request: MaterializeRequest) -> None:
         for feature_view in request.feature_views or []:
-            # TODO: receives a str for resource but isn't in the Union. is str actually allowed?
             assert_permissions(
-                resource=feature_view,  # type: ignore
+                resource=_get_feast_object(feature_view, True),
                 actions=[AuthzedAction.WRITE_ONLINE],
             )
```

Comment:
fix: we don't `assert_permissions` on the str name of the `feature_view`, but on the `FeastObject` itself